project/webapp/model/formatter.js

```diff
@@ -6,14 +6,50 @@ sap.ui.define([], () => {
       this.oResourceBundle = oResourceBundle;
     },
 
+    checkIfItsNewProduct(releaseDate) {
+      if (releaseDate) {
+        const iReleaseDate = new Date(releaseDate).getTime();
+        const iSevenDaysAgo = new Date(Date.now() - 7 * 24 * 60 * 60 * 1000).getTime();
+
+        return iSevenDaysAgo < iReleaseDate;
+      } else {
+        return false;
+      }
+    },
+
+    getDaysFromReleaseDate(sReleaseDate) {
+      if (sReleaseDate) {
+        const iReleaseDate = new Date(sReleaseDate);
+        const oDateNow = new Date();
+        const iDiffInTime = oDateNow.getTime() - iReleaseDate.getTime();
+        const iDiffInDays = Math.round(iDiffInTime / (1000 * 3600 * 24));
+
+        return this.oResourceBundle.getText("DaysText", [iDiffInDays]);
+      }
+    },
+
     formatConfirmMessageText(aProductsNames) {
-      if(!aProductsNames || !aProductsNames.length) {
+      if (!aProductsNames || !aProductsNames.length) {
         return "";
       }
 
       return this.oResourceBundle.getText(aProductsNames.length > 1 ? "ConfirmDeleteProductsText" : "ConfirmDeleteProductText", [
         aProductsNames.length > 1 ? aProductsNames.length : aProductsNames[0]
       ]);
+    },
+
+    formatProdiuctCategories(aCategories) {
+      if (aCategories) {
+        const aCategoriesNames = aCategories.map((item) => item.name);
+
+        return aCategoriesNames.join(" | ");
+      }
+    },
+
+    getCategoriesId(aCategories) {
+      if (aCategories) {
+        return aCategories.map((item) => item.id);
+      }
```

Comment:
in some easy expressions, it would be nice to use the ternary operator
return aCategories?.length ? aCategories.map((item) => item.id) : [];

Also, it would be nice if you would try to involve Optional chaining (?.) in your code.